File: src/main/java/kr/co/lunatalk/domain/auth/dto/response/KakaoUserInfoResponse.java
package kr.co.lunatalk.domain.auth.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;

public record KakaoUserInfoResponse(
	Long id,
	@JsonProperty("kakao_account")
	KakaoAccount kakaoAccount
) {
	public record KakaoAccount(
		String email,
		Profile profile
	) {
		public record Profile(
			String nickname,
			@JsonProperty("profile_image_url")
			String profileImageUrl
		) {
		}
	}
}

File: src/main/kotlin/kr/co/lunatalk/domain/auth/dto/response/KakaoUserInfoResponse.kt
package kr.co.lunatalk.domain.auth.dto.response

import com.fasterxml.jackson.annotation.JsonProperty

data class KakaoUserInfoResponse(
    val id: Long?,
    @JsonProperty("kakao_account")
    val kakaoAccount: KakaoAccount?,
) {
    data class KakaoAccount(
        val email: String?,
        val profile: Profile?,
    ) {
        data class Profile(
            val nickname: String?,
            @JsonProperty("profile_image_url")
            val profileImageUrl: String?,
        )
    }
}
